### PR TITLE
Read in Package.toml and add "repo" to PkgEntry

### PR DESCRIPTION
**Pull request by fchorney**
I'd like to know what the repo path is for each repository so I read in the `Package.toml` to get this information.

### CODE REVIEW

File: src/registry_instance.jl [hunk 1]
```diff
@@ -14,10 +14,21 @@ function RegistryInstance(path::AbstractString)
...
         pkgpath = info["path"]
-        pkg = PkgEntry(pkgpath, path, name, uuid)
+
+        p_repo = ""
+        try
+            p = parsefile(joinpath(path, pkgpath, "Package.toml"))
+            p_name = p["name"]
+            p_uuid = UUID(p["uuid"])
+            p_repo = p["repo"]
+            @assert p_name == name
```
Comment (mattBrzezinski) [inline src/registry_instance.jl]: Not sure I follow the need for assertions here and the try/catch?

Comment (fchorney) [reply src/registry_instance.jl]: The try catch is necessary due to the tests. The `Foobar` registry contains `DownDep` which doesn't actually exist in file/folder form so when trying to read the `Package.toml` file it doesn't exist. 

The asserts are just to make sure the data in Package.toml matches what we see in the Regitry.toml. Unsure if there is a better way to confirm that than to just check for equality.

Comment (mattBrzezinski) [reply src/registry_instance.jl]: - Can we not modify the tests to include this file and bin the `try/catch`?
- I think it's safe to assume that registry files will match each other as this is all an automated process when registering (or at least should be).

Comment (fchorney) [reply src/registry_instance.jl]: sure makes sense to me.

**Review by mattBrzezinski — APPROVED** [review]

**Pull request merged**